client/app/visualizations/vega/Editor.jsx

```diff
@@ -0,0 +1,269 @@
+import React from 'react';
+import { Form, Select, message, Icon } from 'antd';
+import { debounce } from 'lodash';
+// import stringify from 'json-stringify-pretty-compact';
+// import * as YAML from 'js-yaml';
+import AceEditor from 'react-ace';
+import { UndoManager, EditSession } from 'brace';
+
+import { EditorPropTypes } from '../index';
+import { Mode, THEMES, THEME_NAMES, DEFAULT_OPTIONS } from './consts';
+import { renderInitialSpecText, parseSpecText, applyTheme, dumpSpecText } from './helpers';
+
+// Initialize editor configuration (language support, etc.)
+import '@/components/editor';
+
+// Monaco diagnostics option
+//
+// const vegaSchema = require('vega/build/vega-schema.json');
+// const vegaLiteSchema = require('vega-lite/build/vega-lite-schema.json');
+//
+// export const MONACO_SCHEMAS = [
+//   {
+//     fileMatch: [`${Mode.Vega}.*`],
+//     schema: vegaSchema,
+//     uri: 'https://vega.github.io/schema/vega/v5.json',
+//   },
+//   {
+//     fileMatch: [`${Mode.VegaLite}.*`],
+//     schema: vegaLiteSchema,
+//     uri: 'https://vega.github.io/schema/vega-lite/v3.json',
+//   },
+// ];
+// const monacoDiagnostics = {
+//   allowComments: false,
+//   enableSchemaRequest: false,
+//   validate: true,
+//   schemas: MONACO_SCHEMAS,
+// };
+
+// const jsonFormatter = {
+//   provideDocumentFormattingEdits(model) {
+//     return [
+//       {
+//         range: model.getFullModelRange(),
+//         text: stringify(JSON.parse(model.getValue())),
+//       },
+//     ];
+//   },
+// };
+
+// const yamlFormatter = {
+//   provideDocumentFormattingEdits(model) {
+//     return [
+//       {
+//         range: model.getFullModelRange(),
+//         text: YAML.safeDump(YAML.safeLoads(model.getValue())),
+//       },
+//     ];
+//   },
+// };
+
+/**
+ * Add additional language support for the text editor
+ */
+// function setupEditor() {
+//   monaco.languages.json.jsonDefaults.setDiagnosticsOptions(monacoDiagnostics);
+//   monaco.languages.registerDocumentFormattingEditProvider('json', jsonFormatter);
+//   monaco.languages.registerDocumentFormattingEditProvider('yaml', yamlFormatter);
+// }
```

Comment:
Keeping the code for Monaco here for now, so later migration will be easier (if we are going to do it)